hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanPlugin.groovy

```diff
@@ -216,16 +231,32 @@ class AzkabanPlugin implements Plugin<Project> {
    */
   AzkabanProject readAzkabanProject(Project project) {
     def pluginJson = readAzkabanPluginJson(project);
-    if (pluginJson == null) {
-      throw new GradleException("\n\nPlease run \"gradle writeAzkabanPluginJson\" to create a default .azkabanPlugin.json file in your project directory which you can then edit.\n")
+    AzkabanProject azkProject = makeDefaultAzkabanProject(); //changed makeAzkabanProject here to makeDefaultAzkabanProject()
+
+    if (pluginJson != null) {
+      // If the file exists, the task should use this information, but give the user the chance
+      // to confirm or change the Azkaban URL / project / user. If the user changes this information,
+      // ask them if they want to save the changes (to the .azkabanPlugin.json file).
+
+      azkProject.azkabanProjName = pluginJson[AZK_PROJ_NAME];
+      azkProject.azkabanUrl = pluginJson[AZK_URL];
+      azkProject.azkabanUserName = pluginJson[AZK_USER_NAME];
+      azkProject.azkabanValidatorAutoFix = pluginJson[AZK_VAL_AUTO_FIX];
+      azkProject.azkabanZipTask = pluginJson[AZK_ZIP_TASK];
+    } else {
+      // The file doesn't exist, the azkabanUpload task should ask the user for this information.
+      println("File .azkabanPlugin.json not found. Automatically switching to Interactive mode");
```

Comment:
Use logger.lifecycle instead of println to ensure the message prints atomically instead of possibly being cut into pieces